src/modules/avformat/producer_avformat.c

```diff
@@ -2806,8 +2865,10 @@ static void producer_set_up_video(producer_avformat self, mlt_frame frame)
     // Fetch the video format context
     AVFormatContext *context = self->video_format;
 
-    // Get the video_index
-    int index = mlt_properties_get_int(properties, "video_index");
+    // Get the video stream index
+    int index = mlt_properties_get(properties, "vstream")
+                    ? mlt_properties_get_int(properties, "vstream")
```

Comment:
Should this be:

```
absolute_stream_index(self->video_format,
                                      AVMEDIA_TYPE_VIDEO,
                                      mlt_properties_get_int(properties, "vstream")
```

?

Reply:
No, at this point `index` is not qualified as relative or absolute. And to convert relative to absolute it uses the AVFormatContext, which could be NULL. So, at this point, we only need to see if the stream is disabled to skip opening the context. Otherwise, the context is reopened if needed, and then it can use `pick_video_stream()` to convert a relative stream index to absolute.